src/Session.cc

```diff
@@ -560,9 +560,19 @@ const AddressSpace::Mapping Session::recreate_shared_mmap(
   uint32_t flags = m.flags;
   size_t size = m.map.size();
   void* preserved_data = preserve == PRESERVE_CONTENTS ? m.local_addr : nullptr;
-  if (preserved_data) {
-    remote.task()->vm()->detach_local_mapping(m.map.start());
-  }
+
+  // Note that Mapping `m` may correspond to a Mapping from a different Task than
+  // the `maybe_detach_mapping`. See replay_syscall.cc prepare_clone() for an example.
+  auto maybe_detach_mapping = remote.task()->vm()->mapping_of(m.map.start());
```

Comment:
Call this "remote_task_mapping"; put it in a block with the two asserts to make it clear this is only for the asserts, we don't use it otherwise.

Reply:
Done